test/hpr_config_worker_SUITE.erl

```diff
@@ -0,0 +1,122 @@
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%   Running tests for this suite
+%% @end
+%%--------------------------------------------------------------------
+all() ->
+    [
+        full_test
+    ].
+
+%%--------------------------------------------------------------------
+%% TEST CASE SETUP
+%%--------------------------------------------------------------------
+init_per_testcase(TestCase, Config) ->
+    %% Startup a config service test server
+    _ = application:ensure_all_started(grpcbox),
+    {ok, ServerPid} = grpcbox:start_server(#{
+        grpc_opts => #{
+            service_protos => [config_pb],
+            services => #{'helium.config.config_service' => test_config_service}
+        },
+        listen_opts => #{port => ?PORT, ip => {0, 0, 0, 0}}
+    }),
+
+    %% Setup config worker
+    BaseDir = erlang:atom_to_list(TestCase) ++ "_data",
+    application:set_env(hpr, config_worker, #{
+        host => "localhost",
+        port => ?PORT,
+        dets_backup_path => filename:join(BaseDir, "config_worker.dets")
+    }),
+
+    test_utils:init_per_testcase(TestCase, [{server_pid, ServerPid} | Config]).
+
+%%--------------------------------------------------------------------
+%% TEST CASE TEARDOWN
+%%--------------------------------------------------------------------
+end_per_testcase(TestCase, Config) ->
+    meck:unload(),
+    test_utils:end_per_testcase(TestCase, Config),
+    ok = gen_server:stop(proplists:get_value(server_pid, Config)),
+    application:set_env(hpr, config_worker, #{}),
+    ok.
+
+%%--------------------------------------------------------------------
+%% TEST CASES
+%%--------------------------------------------------------------------
+
+full_test(_Config) ->
+    %% Let it startup
+    timer:sleep(100),
+
+    %% Create route and send them from server
+    Route1 = #config_route_v1_pb{
+        net_id = 0,
+        devaddr_ranges = [
+            #config_devaddr_range_v1_pb{start_addr = 16#00000001, end_addr = 16#0000000A}
+        ],
+        euis = [#config_eui_v1_pb{app_eui = 1, dev_eui = 0}],
+        oui = 1,
+        protocol = {router, #config_protocol_router_pb{ip = <<"localhost">>, port = 8080}}
+    },
+    Route2 = #config_route_v1_pb{
+        net_id = 0,
+        devaddr_ranges = [
+            #config_devaddr_range_v1_pb{start_addr = 16#00000010, end_addr = 16#0000001A}
+        ],
+        euis = [#config_eui_v1_pb{app_eui = 2, dev_eui = 2}],
+        oui = 2,
+        protocol = {gwmp, #config_protocol_gwmp_pb{ip = <<"localhost">>, port = 8088}}
+    },
+    Routes = [Route1, Route2],
+    ConfigRouteResV1 = #config_routes_res_v1_pb{routes = Routes},
+    ok = test_config_service:config_route_res_v1(ConfigRouteResV1),
+
+    %% Let  process new routes
+    timer:sleep(100),
+
+    %% Check Dets
+    {state, _Host, _Port, _Conn, _Stream, _Path, Dets} = sys:get_state(hpr_config_worker),
```

Comment:
This is kind of ugly. Though I guess we'll get a match error if the state record changes. Since we know the name of the dets file, can the test open the file instead of reusing the dets handle from the worker?